crates/telemetry/src/lib.rs

```diff
@@ -15,18 +15,86 @@ limitations under the License.
 */
 
 use meter::METER;
-use opentelemetry::metrics::Counter;
-use std::sync::LazyLock;
+use opentelemetry::{
+    metrics::{Counter, Histogram},
+    KeyValue,
+};
+use std::{
+    sync::{Arc, LazyLock},
+    time::Duration,
+};
 
 #[cfg(feature = "anonymous_telemetry")]
 pub mod anonymous;
 mod exporter;
 mod meter;
 
-pub static QUERY_COUNT: LazyLock<Counter<u64>> = LazyLock::new(|| {
+static QUERY_COUNT: LazyLock<Counter<u64>> = LazyLock::new(|| {
     METER
         .u64_counter("query_count")
         .with_description("Number of queries run.")
         .with_unit("queries")
         .init()
 });
+
+pub fn track_query_count() {
+    QUERY_COUNT.add(1, &[]);
+}
+
+static BYTES_SCANNED: LazyLock<Counter<u64>> = LazyLock::new(|| {
```

Comment:
Learning Question: Could you explain why the LazyLock is used here? It looks like there are several nested layers of LazyLock within LazyLock::new(|| METER). Is it for hierarchical lazy initialization?

Reply:
Yeah, its so that each metric is only initialized when its first used, and the first metric to be used will also lazily initialize the METER itself.